pkg/manager/manager.go

```diff
@@ -348,6 +348,41 @@ func (m *Manager) computeChange(updates types.TypedValueMap,
 	return change.NewChange(newChanges, description)
 }
 
+// ComputeNewDeviceChange computes a given device change the given updates and deletes, according to the path
+// on the configuration for the specified target
+func (m *Manager) ComputeNewDeviceChange(deviceName string, version string,
+	deviceType string, updates devicechangetypes.TypedValueMap,
+	deletes []string, description string) (*devicechangetypes.Change, error) {
+
+	var newChanges = make([]*devicechangetypes.ChangeValue, 0)
+	//updates
+	for path, value := range updates {
+		changeValue, err := devicechangetypes.NewChangeValue(path, value, false)
+		if err != nil {
+			log.Warningf("Error creating value for %s %v", path, err)
+			continue
+		}
+		newChanges = append(newChanges, changeValue)
+	}
+	//deletes
+	for _, path := range deletes {
+		changeValue, _ := devicechangetypes.NewChangeValue(path, devicechangetypes.NewTypedValueEmpty(), true)
+		newChanges = append(newChanges, changeValue)
+	}
+	//description := fmt.Sprintf("Originally created as part of %s", description)
```

Comment:
Probably time to dump the `description`. It will now always be traceable back to the network change, which will have the description